src/hypertable_cache.c

```diff
@@ -129,24 +130,55 @@ ts_hypertable_cache_invalidate_callback(void)
 
 /* Get hypertable cache entry. If the entry is not in the cache, add it. */
 TSDLLEXPORT Hypertable *
-ts_hypertable_cache_get_entry(Cache *cache, Oid relid)
+ts_hypertable_cache_get_entry(Cache *const cache, const Oid relid, const bool missing_ok)
 {
+	Hypertable *ht;
+
 	if (!OidIsValid(relid))
-		return NULL;
+		if (missing_ok)
+			return NULL;
+		else
+			ereport(ERROR, (errcode(ERRCODE_UNDEFINED_OBJECT), errmsg("invalid Oid")));
+	else
+		ht = ts_hypertable_cache_get_entry_with_table(cache, relid, NULL, NULL);
```

Comment:
I don't understand why it is blocking request, since it is a personal taste and is not followed in our code. And I see your code, which doesn't use braces in multiline code.

Reply:
OK, the comment is about the then-statement to cover `if(missing_ok)`. I will fix it.

Reply:
Fixed